packages/debugger/src/panels/sources/model.ts

```diff
@@ -17,8 +21,43 @@ export class SourcesModel implements IDebugger.Model.ISources {
    */
   constructor(options: SourcesModel.IOptions) {
     this.currentFrameChanged = options.currentFrameChanged;
+    this._currentSource = null;
+    this._currentFrame = null;
+
     this._displayRegistry =
       options.displayRegistry ?? new DebuggerDisplayRegistry();
+
+    if (options.editorServices?.mimeTypeService) {
+      this._mimeTypeService = options.editorServices?.mimeTypeService;
+    }
+
+    this.currentFrameChanged.connect(async (_, frame) => {
+      if (frame) {
```

Comment:
We should probably update the frame and source anyway (the suggestion below can't be commit as it, it would fail).

```suggestion
      this.currentFrame = frame;
      if (!frame) {
        this.currentSource = null;
        return;
      }
```